src/openassetio-core/src/errors/exceptionMessages.cpp

```diff
@@ -47,16 +47,24 @@ std::string createBatchElementExceptionMessage(const BatchElementError& err, siz
    * Ends up looking something like : "entityAccessError: Could not
    * access Entity [index=2] [access=read] [entity=bal:///entityRef]"
    */
-  const auto errorCodeStr = fmt::format("{}:", errorCodeName(err.code));
-  const auto errorMessageStr = err.message.empty() ? "" : fmt::format(" {}", err.message);
+  std::string result;
 
-  // Data elements
-  const auto indexStr = fmt::format(" [index={}]", index);
-  const auto accessStr = fmt::format(" [access={}]", access::kAccessNames[access]);
-  const auto entityReferenceStr = fmt::format(" [entity={}]", entityReference.toString());
+  result += fmt::format("{}:", errorCodeName(err.code));
 
-  return fmt::format("{}{}{}{}{}", errorCodeStr, errorMessageStr, indexStr, accessStr,
-                     entityReferenceStr);
+  if (!err.message.empty()) {
+    result += " ";
+    result += err.message;
+  }
+
+  result += fmt::format(" [index={}]", index);
+
+  if (access) {
+    result += fmt::format(" [access={}]", access::kAccessNames[*access]);
+  }
+
+  result += fmt::format(" [entity={}]", entityReference.toString());
+
+  return result;
```

Comment:
[Consider][Minor] Seems a pretty big stylistic change when as far as I can tell you could have just continued the pre-existing, (and imo cleaner) pattern by putting a ternery on access. Unless i'm missing other functionality changes.

Reply:
It felt like the additional conditional pushed it over a line re. readability making it clear that `access` and `err.message` might not be included in the message. A ternary kinda buries the condition, so that when you see the final string construction its not obvious that some placeholders may be blank.